### PR TITLE
chore: tune dependabot for processing requirements

### PR DESCRIPTION
**Pull request by Bento007**
## Reason for Change

- This reduces the number of dependabot PR that will be open for the processing container

## Changes

- remove unused dependencies from processing requirements.txt
- add rule to group minor and patch updates
- group database dependencies together.

## Testing steps

- verify unit test pass

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -55,6 +55,24 @@ updates:
       day: "sunday"
     assignees:
       - "Bento007"
+    groups:
+      database:
```
Comment (Bento007) [inline .github/dependabot.yml:59]: FYI: this list is ordered from top to bottom. Dependabot will open a PR against the first group that matches.

File: python_dependencies/processing/requirements.txt [hunk 2]
```diff
@@ -1,23 +1,17 @@
...
 anndata==0.8.0
 awscli
 boto3>=1.11.17
 botocore>=1.14.17
 cellxgene-schema
 dataclasses-json
 ddtrace==2.1.4
-furl
-jsonschema
-matplotlib==3.7.3
```
Comment (Bento007) [inline python_dependencies/processing/requirements.txt:11]: The only place matplotlib is used in the code is in **wmg_performance_profiling.py**

File: python_dependencies/processing/requirements.txt [hunk 3]
```diff
@@ -1,23 +1,17 @@
-alembic>=1.9, <2
```
Comment (Bento007) [inline python_dependencies/processing/requirements.txt:1]: this is only needed for running a migrations or creating the database, none of which are done by the processing container.

File: python_dependencies/processing/requirements.txt [hunk 4]
```diff
@@ -1,23 +1,17 @@
...
 awscli
 boto3>=1.11.17
 botocore>=1.14.17
 cellxgene-schema
 dataclasses-json
 ddtrace==2.1.4
-furl
-jsonschema
-matplotlib==3.7.3
-moto>=5.0.0
```
Comment (Bento007) [inline python_dependencies/processing/requirements.txt:12]: only needed for testing

File: python_dependencies/processing/requirements.txt [hunk 5]
```diff
@@ -1,23 +1,17 @@
-alembic>=1.9, <2
 anndata==0.8.0
 awscli
 boto3>=1.11.17
 botocore>=1.14.17
 cellxgene-schema
 dataclasses-json
 ddtrace==2.1.4
-furl
```
Comment (Bento007) [inline python_dependencies/processing/requirements.txt:9]: only needed for testing

File: python_dependencies/processing/requirements.txt [hunk 6]
```diff
@@ -1,23 +1,17 @@
...
-jsonschema
```
Comment (Bento007) [inline python_dependencies/processing/requirements.txt:10]: It doesn't appear to be used anywhere in the code

File: python_dependencies/processing/requirements.txt [hunk 7]
```diff
@@ -1,23 +1,17 @@
...
-matplotlib==3.7.3
-moto>=5.0.0
 numba==0.56.2
 numpy==1.23.2
 pandas==1.4.4
 psutil>=5.9.0
 psycopg2-binary>=2.8.5
 pyarrow>=1.0
 pydantic>=1.9.0
-PyMySQL==0.9.3
```
Comment (Bento007) [inline python_dependencies/processing/requirements.txt:20]: It doesn't appear to be used anywhere in the code.

**Review by nayib-jose-gloria — APPROVED** [review]

**Pull request merged**